### PR TITLE
Update header links in UG and fix bugs

### PR DESCRIPTION
**Pull request by jacobkwan**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -130,7 +130,7 @@ This section describes important information for you to take note of before and
 * An example is shown below:
 ![SortExample](images/SortExample.png)
 
-#### Responsive UI when resizing window
+### 4.4 Responsive UI when resizing window
```
Comment (rylzxc) [inline docs/UserGuide.md:133]: gd catch!

File: docs/UserGuide.md [hunk 2]
```diff
@@ -246,7 +246,7 @@ Other examples:
 
 ### 5.3. Listing all existing internship applications
 
-Lists all of your internship applications in WorkBook in a [sorted order](#3-things-to-note).
+Lists all of your internship applications in WorkBook in a [sorted order](#43-behaviour-of-sorted-internship-applications).
```
Comment (rylzxc) [inline docs/UserGuide.md:249]: 43 == 4.3 isit?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -269,13 +269,13 @@ Below are the application stages that we have included tips for.
 | `Phone Interview`      |
 | `Rejected`             |
 
-If WorkBook is in the [narrow layout](#responsive-ui-when-window-resizing), simply click on the light bulb icon to view the tips for that application. A window will appear showing all the tips for that application.
+If WorkBook is in the [narrow layout](#44-responsive-ui-when-resizing-window), simply click on the light bulb icon to view the tips for that application. A window will appear showing all the tips for that application.
```
Comment (rylzxc) [inline docs/UserGuide.md:272]: ok i assume it is

**Review by rylzxc — APPROVED** [review]
lgtm

File: docs/UserGuide.md (hunk 2 above)
Comment (jacobkwan) [reply docs/UserGuide.md:249]: yup thats the way the ids are generated for markdown headers

**Pull request merged**